uiowa/uiowa
# Commencement updates

# How to test
- You might want to start by opening a second command line instance and running the command below related to checking facilities because it takes a while and can run in the background until done.
- Code review
- After pulling locally, perform a find in files search (right click on the top level directory) of the entire project for `field_event_start_date` and `field_event_end_date` to make sure there are no longer any references to them.
```
ddev blt ds --site commencement.uiowa.edu && ddev drush @commencement.local import-events && ddev drush @commencement.local uli admin/content
```
- Spot check some events
- Ensure date(s), time(s), and duration are correct. Duration should be 0 on an event with no end date/time.
- Confirm that dates appear on the event node.
- Add some additional information to a Venue and check the markup to ensure expected aria output. `aria-dataparent=''` id should match the `id` of the accordion container.

## Check existing importers for regressions

### Classrooms

```
ddev blt ds --site classrooms.uiowa.edu && ddev drush @classrooms.local classrooms-rooms && ddev drush @classrooms.local classrooms-buildings
```
- Confirm no errors during processing. Spot check rooms and buildings.

### Emergency

```
ddev blt ds --site emergency.uiowa.edu && ddev drush @emergency.local rave-alerts
```
- Confirm no errors during processing.

### Facilities

This update includes an enhancement the EntityItemProcessorBase class that allows mapping individual fields from a source record to parts of a multi-part field. As such, we need to ensure that I didn't break the existing implementations.

- Unfortunately, this next command will take 15+ minutes to run at least.

```
ddev blt ds --site facilities.uiowa.edu && ddev drush @facilities.local fm-projects && ddev drush @facilities.local fm-buildings
```
- Confirm both imports run without errors.
- Spot check some building and projects that were imported to ensure there were no changes.
